cli/src/commands/branch/delete.rs

```diff
@@ -47,8 +47,6 @@ pub fn cmd_branch_delete(
             .set_local_branch_target(branch_name, RefTarget::absent());
     }
     tx.finish(ui, format!("delete {}", make_branch_term(&names)))?;
-    if names.len() > 1 {
```

Comment:
I think the point of these is not to avoid verbosity, but as an alert. 90% of the time, when a person uses `jj branch delete`, they expect to delete one branch. The message would tell them something went wrong.

It's much easier to notice a message that's usually absent, as opposed to the difference between "Deleted 1 branches" and "deleted 2 branches".

Reply:
Perhaps we should add a comment explaining these are meant as an alert in case of a mistake?

Reply:
We could, in principle, print "Deleted 1 branch" in neutral color, and "Deleted **2 branches**" with red highlighting, but I think that would go against your motivation for this change. I don't think that would be better than not printing "Deleted 1 branch".

Reply:
Hmm, I disagree about the absence of "Deleted 1 branch." I think it's nice to see a short summary describing what's changed, as we print "Rebased N commits" and checkout stats.

Maybe we can add a warning if `args.names.len() != matched_branches.len()` (in addition to stats)?

Reply:
I'm trying to remember when this was most useful. One case I can remember is:

```
jj branch create branch xyz
```

where you forgot `-r` before the xyz. I'll try to remember if there were other cases. Here, your suggestion wouldn't help, but we can probably come up with something else.

Reply:
I see you predicted this and didn't make this change to `create` or `set`, actually.

Reply:
> jj branch create branch xyz

I see. In that case, the warning should be displayed if `matched_branches.len() > 1`.

For `set`/`create`/`move`, it might be better to show `branch list`-like output as stats, but I haven't considered that seriously.

Reply:
How much would you mind making the user-facing message use the singular "Deleted 1 branches" or "Deleted branch XXXX"? I know it complicates the code back again, but I think it makes the tool sound friendlier.

With the same code complication, we could also split out the UI changes to a separate PR, I think.

The other thing I was thinking is to add comments explaining why we show this message for 1 branch in some commands and not others.

Depending on how strongly you feel, I can approve this, none of my concerns seem like absolute deal-breakers, but I'm still vaguely uncomfortable about setting these UI changes in stone. Or we can change them later, I suppose. The other parts of this PR seem uncontroversial and very helpful.

Reply:
> How much would you mind making the user-facing message use the singular "Deleted 1 branches" or "Deleted branch XXXX"? I know it complicates the code back again, but I think it makes the tool sound friendlier.

I'd rather want to do that globally by using some i18n library. At this point, I'm against introducing local ad-hoc changes.

BTW, it might be okay to use always singular "branch" with a list of branch names.
(i.e. `Deleted branch XXX, YYY, ..`.)

WDYT? It also applies to transaction description.

> With the same code complication, we could also split out the UI changes to a separate PR, I think.

Yeah, I'm thinking about it. I just include it because I didn't expect ithis could be a blocker.

Reply:
Removed UI change from this PR, and changed transaction description to use singular "branch". PTAL.

Reply:
Thank you very much!

To finish up some loose ends on the discussion:

There is also an option of `Deleted X branch(es)`. It's not super-exciting, but I should mention it for completeness.

> Deleted branch XXX, YYY, ..

I don't think this helps, unfortunately. If one were writing naturally, it would be "Deleted branches XXX, YYY, and ZZZ". There is also the question of what to do if there are many branches, just list them all?

The following still looks unnatural, but a little better to me in terms of grammar, since it doesn't try to be a sentence (I wonder if native English speakers would agree):

> Branches deleted: 1

or

> Branches deleted: XXX (1)